lib/kitchen/driver/ec2.rb

```diff
@@ -354,6 +361,8 @@ def hostname(server, interface_type = nil)
           potential_hostname = nil
           INTERFACE_TYPES.values.each do |type|
             potential_hostname ||= server.send(type)
+            # AWS returns an empty string if the dns name isn't populated yet
```

Comment:
`private_ip` is the most likely to be available but not the most likely to be reachable by SSH.  The logic here is currently

```ruby
# if `interface_type` is provided:
  # query only that interface_type
# else
  # first, try dns
  # second, try public_ip
  # third, try private_ip
  # return the first value that is non-nil and non-empty
```

This brings the `hostname` logic in line with what is in the [README](https://github.com/test-kitchen/kitchen-ec2/blob/aa765caf608efd07040a527cf2cfcb05daa3f87b/README.md#interface).  I don't want to break the backwards compatibility with logic that has been there for a while.

I also like the 'try from easiest to SSH into to hardest to SSH into' logic rather than 'try from what is most likely to be available to what is least likely to be available'.  The goal is to get a hostname we can SSH into and then install Chef.  If a user really wants to use private_ip they can specify `interface: private` in their .kitchen.yml.